_posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md

```diff
@@ -0,0 +1,361 @@
+--payload="$(base64 < "$certificate")" --expiration $(date --date="$(openssl x509 -enddate -noout \
+-in "$certificate"|cut -d= -f 2)" --iso-8601)
+```
+Make sure to save the returned secret_href as variable, we need that later
+```bash
+intermediates_url="https://keymanager.domain.tld:/v1/secrets/uuid"
+```
+
+**Step 6**
+Create a certificate container containing the certificate, all intermediates
+the passphrase and the private key.
+
+```bash
+name="${domain}_container"
+openstack secret container create --name "${name}" --type certificate \
+-s "certificate=${certificate_url}" -s "intermediates=${intermediates_url}" \
+-s "private_key=${private_key_url}" -s "private_key_passphrase=${passphrase_url}"
+
+```
+Make sure to save the returned secret_href as variable, we need that later
+```bash
+octavia_certificate_url="https://keymanager.domain.tld:/v1/containers/uuid"
+```
+
+### Uploading the SSL certificate to keymanager as single file
+The alternative way to store a certificate to use with an Octavia loadbalancer is to create a single
+pkcs12 file with all certificates without password protection.
+The downside of this approach is we need to temporarily store the unencrypted private key for the
+load balancer on the OpenStack CLI server. The benefit is we can select the certificate in Horizon.
+
+
+**Prerequisites**
+ - The server certificate, intermediate certificates and private key are stored in the proper order in
+a single file on the OpenStack CLI server named 'certificate.pem'
+ - openssl tooling installed on the OpenStack CLI server
+
+**Step 1**  
+First make sure you have setup the OpenStack CLI and that you are able to
+execute commands using the `openstack` command. For more information please
+refer to the
+[Using the OpenStack CLI article](
+    {{ '/articles/using-the-cli-linux' | relative_url }}
+).
+
+**Step 2**
+Convert the certificates to a pkcs12 certificate (skip this if you already have a pkcs12 encoded
+file with all required certificates):
+```bash
+openssl pkcs12 -export -inkey private.key -in certificate.pem -certfile intermediate.pem \
+-passout pass: -out complete.p12
+```
+
+**Step 3**
+
+Store the certificate in barbican
+```bash
+certificate=complete.p12
+domain="$(openssl pkcs12 -in "$certificate" -nokeys -passin pass: | openssl x509 -noout \
+-subject | cut -d= -f 3| tr -d ' ')"
+name="${domain}_complete_certificate"
+expiration_date="$(date --date="$(openssl pkcs12 -in "$certificate" -nokeys -passin pass: | \
+openssl x509 -enddate -noout | cut -d= -f 2)" --iso-8601)"
+openstack secret store --name="${name}" -t 'application/octet-stream' -e 'base64' \
+--payload="$(base64 < "$certificate")" --expiration "${expiration_date}"
+```
+Make sure to save the returned secret_href as variable, we need that later
+```bash
+octavia_certificate_url="https://keymanager.domain.tld:/v1/secrets/uuid"
+```
+
+---
+
+## Creating the loadbalancer
```

Comment:
Maybe add line breaks behind the steps so the style of the steps is the same throughout the article (Optional)